app/models/order_cycle.rb

```diff
@@ -222,7 +222,7 @@ def exchanges_carrying(variant, distributor)
   end
 
   def exchanges_supplying(order)
-    exchanges.supplying_to(order.distributor).with_any_variant(order.variants)
+    exchanges.supplying_to(order.distributor).with_any_variant(order.variants.map(&:id))
```

Comment:
Same here. I'm not sure what performs better at the moment but I would guess that `select` scales better than `map`, especially because we use the result in sub queries and the database can handle it very efficiently.

Reply:
this would not be a subquery, order.variants are already loaded at this point. so map is better.

Reply:
so, if order.variants is already loaded, AR will not make it a subquery. and using select on an already loaded order.variants will produce an extra query. probably because of the intense chaining of scopes...